signer/src/block_observer.rs

```diff
@@ -121,18 +121,38 @@ where
         // early.
         let run = async {
             while let Some(new_block_hash) = self.bitcoin_blocks.next().await {
-                self.load_latest_deposit_requests().await?;
+                if let Err(error) = self.load_latest_deposit_requests().await {
+                    tracing::warn!(%error, "could not load latest deposit requests from Emily");
+                }
 
                 // TODO: What to do when `new_block_hash?` errors? Perhaps we can
                 // handle this within a failover-stream if this indicates a problem
                 // with the stream, and then we change this back to a plain `BlockHash`
                 // instead of a `Result<>`.
-                for block in self.next_blocks_to_process(new_block_hash?).await? {
-                    self.process_bitcoin_block(block).await?;
+                match new_block_hash {
```

Comment:
nit: we could save a couple of indentations to improve readability here with a couple of early returns (continues)

Reply:
+1 on the early returns, but for the sake of time I'd settle for this in the comment above:

```
// TODO: Return early to save indentations in the following code.
```

Reply:
I took a quick crack at it, thoughts on this:

```rust
        // TODO: We need to revisit all of the `?`'s in this function to ensure
        // that we don't accidentally kill the signer by exiting this function
        // early.
        let run = async {
            while let Some(maybe_new_block_hash) = self.bitcoin_blocks.next().await {
                // TODO: What to do when `new_block_hash?` errors? Perhaps we can
                // handle this within a failover-stream if this indicates a problem
                // with the stream, and then we change this back to a plain `BlockHash`
                // instead of a `Result<>`.
                self.load_latest_deposit_requests().await
                    .inspect_err(|error| tracing::warn!(%error, "could not load latest deposit requests from Emily"))?;
                let new_block_hash = maybe_new_block_hash
                    .inspect_err(|error| tracing::warn!(%error, "could not get next blocks to process"))?;
                tracing::info!(%new_block_hash, "observed new bitcoin block from stream");
                let next_blocks_to_process = self.next_blocks_to_process(new_block_hash).await
                    .inspect_err(|error| tracing::warn!(%error, "could not get next blocks to process"))?;
                for block in next_blocks_to_process {
                    if let Err(error) = self.process_bitcoin_block(block).await {
                        tracing::warn!(%error, "could not process bitcoin block");
                    }
                }
                self.context.signal(SignerEvent::BitcoinBlockObserved.into())?;
            }

            Ok::<_, Error>(())
        };
```

Reply:
Haha, had zero focus on formatting for this PR. Refactored in [048b329](https://github.com/stacks-network/sbtc/pull/634/commits/048b32963283725d4555891d4310a1c44291d348)

Reply:
Lol, sorry for the nits!